channeldb/graph.go

```diff
@@ -2292,6 +2292,49 @@ func nodeTraversal(tx *bbolt.Tx, nodePub []byte, db *DB,
 		copy(nodeStart[:], nodePub)
 		copy(nodeStart[33:], chanStart[:])
 
+		fetchChannel := func(chanID []byte) (*ChannelEdge, error) {
+
+			intChanID := byteOrder.Uint64(chanID)
+
+			// db isn't assured to be present sadly
```

Comment:
I think a recently merged PR makes this removable. I had these changes wip for a while 😬

Reply:
This function is called by `LightningNode.ForEachChannel`, which is in turn called from a quite a few places. Running this code in production I had a few seg faults due to `db` here. I couldn't quite figure out where the problem was coming from. There's quite a few places where `LightningNode` structs are created without setting `db` that could be causing those crashes.

Reply:
So it cannot be removed despite the recently merged pr that you mentioned? This also means that `LightningNode.ForEachChannel` is not always querying the cache? Is it possible to move this `if db == nil` exception further up the call tree? I don't think you want your pr to sprawl too far, but ideally it shouldn't add any new technical debt in the form of these if statements. `db` is a singleton object that shouldn't be too difficult to obtain anywhere.

To found out where the seg fault comes from, you could just log a stack trace in case the function is entered with `db == nil`.